### DISCUSSION TITLE
Custom Admission Webhook in K8s

### DISCUSSION THREAD

**Discussion by SandraGH5** [flyteorg/flyte#1371] [Q&A] [answered]
How can I use a custom Admission Webhook in K8s to control resource quotas in Flyte?

- Alex Pozimenko

**Comment by SandraGH5** [comment]
Ensure that your Admission webhook returns a Forbidden response with the `exceeded quota` message. The response code can be set by setting `AdmissionResponse.Result.Code` to `403`.

@EngHabu 
@kumare3

**Comment by alexp-lyft** [comment, marked as answer]
Also make sure the webhook returns HTTP `200 OK`. Otherwise the response will be wrapped in `500 InternalServerError` by kube prior to returning to the client

Sample AdmissionReview response with `kind`, `message`, `reason` and `code` set as required:
```
ar.Response.Result = &metav1.Status{
            Message: "... exceeded quota ...",
            Code:    403,
            Reason:  "Forbidden",
}
ar.SetGroupVersionKind(v1.SchemeGroupVersion.WithKind("AdmissionReview")) 
```

**Reply by SandraGH5** [reply]
Thank you Alex!
